process/peer/process.go

```diff
@@ -680,7 +700,7 @@ func (vs *validatorStatistics) setToJailedIfNeeded(
 		return
 	}
 
-	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch && peerAccount.GetList() == string(core.InactiveList) {
+	if vs.isValidatorWithLowRating(peerAccount) {
```

Comment:
&& peerAccount.GetList() == string(core.InactiveList), as it could be at this point already jailed

Reply:
That set does not matter. It is ok right now. In case of waiting or eligible there is an early return